design.md

```diff
@@ -0,0 +1,127 @@
+
+Login screen - This is the initial login screen that the user sees when they open the application. From here, they can enter their username and either attempt to log in or create a new account with the entered username. If the user attempts to create a new account with an existing username, they receive an appropriate error message. Once logged in, the user goes to the Available Tournaments page.
+
+![Available tournaments](https://i.imgur.com/nXL5AIA.png)
+
+Available tournaments - this screen contains a list of available tournaments. Each tournament shows basic details, as well as an indicator if the user is playing in or hosting the tournament. The user is able to sort by date ascending, descending, or put the tournaments they are participating in or hosting at the top. From here, the user is able to go to the tournament creation screen via the top right button or select a tournament to view more details.
+
+
+
+![Tournament creation](https://i.imgur.com/yjCSBzx.png)
+
+Tournament creation - The user can fill out the details of the tournament and create the tournament for players to enter. Once created, the user is taken to the “my tournaments” page, where the newly-created tournament should appear.
+
+![Tournament details](https://i.imgur.com/Kzz9tfd.png)
+
+Tournament details - Lists the details of the tournament, as well as the current scores standings. Players not yet in the tournament are able to join the tournament, and participating players are able to update their score within the tournament or leave the tournament. Standings are updated automatically whenever any player enters a new score. Users are able to look at standings on individual holes via a drop down hole select field. The tournament host is able to delete the tournament.
+
+___
+
+### Data storage
+
+Data is stored in Google Datastore. There are three tables for three different types of objects: users, tournaments, and courses. User objects have the following attributes:
+
+* Username
+* List of tournament IDs the user is hosting
+* List of tournament IDs the user is participating in
+
+Tournament objects have the following attributes:
+
+* Tournament name
+* Tournament ID
+* Hosting user
+* Date
+* Time
+* Entry fee
+* Course ID
+* List of player names, each of which has an associated list of scores
+
+Course objects have the following attributes:
+
+* ID
+* Name
+* Location
+* List of par scores for each hole
+
+The Android application communicates with the datastore via a HTTP server set up with Google App Engine, which acts as an intermediary between the application and the datastore. The application will send HTTP messages to the server to either request data from the datastore or modify data within the datastore, and the server will retrieve or modify data in the datastore accordingly. Datastore does have an API that can be used to access the data directly, which can potentially be used as an alternative to the Google App Engine server.
```

Comment:
is this server stateful or stateless?  meaning,  can this server crash, and and it will function properly after come up.

If million of users using your app,  can multiple servers be running paralleled, and different users will go to different servers.